### DISCUSSION TITLE
[FEATURE REQUEST] Salt should be able to handle Cumulus Linux VRFs (Virtual routing and forwarding)

### DISCUSSION THREAD

**Discussion by Koni-DFX** [saltstack/salt#67288] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

We use Cumulus Linux with Virtual Routing and Forwarding (VRFs). One of the VRFs is the management (mgmt) VRF. The internal Debian mirror, for example, can only be accessed from this mgmt VRF.

Scripts / commands can be executed in the mgmt VRF as follows:

`sudo ip vrf exec mgmt apt-get install snmp-mibs-downloader`

Since salt installs the Debian packages with apt via systemd-run, only the default VRF is used and our Debian mirror is not accessible.

**Describe the solution you'd like**

We use the following patch in salt/modules/aptpkg.py
```
   cmd.extend(["systemd-run", "--scope", "--description", '"{}"'.format(__name__)])
+    vrf = __salt__["config.get"]("systemd.vrf", "")
+    if (vrf):
+        cmd.extend(["/usr/sbin/ip", "vrf", "exec", vrf])
+
     cmd.extend(args)
```

In the minion configuration (/etc/salt/minion.d/local.conf) we define the VRF:

`systemd.vrf: mgmt
`

**Describe alternatives you've considered**

n/a

**Additional context**

[Cumulus Linux - Virtual Routing and Forwarding - VRF.](https://docs.nvidia.com/networking-ethernet-software/cumulus-linux-59/Layer-3/VRFs/Virtual-Routing-and-Forwarding-VRF/)

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://via.vmw.com/salt-slack)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!
